[quant][[fx] update observer_fqn to not depend on node.name

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66878
* __->__ #66767

Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31752052](https://our.internmc.facebook.com/intern/diff/D31752052)

supriyar committed:
[quant][[fx] update observer_fqn to not depend on node.name

Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][[fx] update observer_fqn to not depend on node.name"


Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jingsh reviewed (approved):
OSS changes looks good

There's some fb tests failing on D31752052

supriyar committed:
Update on "[quant][[fx] update observer_fqn to not depend on node.name"


Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31752052](https://our.internmc.facebook.com/intern/diff/D31752052)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][[fx] update observer_fqn to not depend on node.name"


Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31752052](https://our.internmc.facebook.com/intern/diff/D31752052)

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31752052).

supriyar committed:
Update on "[quant][[fx] update observer_fqn to not depend on node.name"


Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31752052](https://our.internmc.facebook.com/intern/diff/D31752052)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][[fx] update observer_fqn to not depend on node.name"


Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31752052](https://our.internmc.facebook.com/intern/diff/D31752052)

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31752052).

supriyar committed:
Update on "[quant][[fx] update observer_fqn to not depend on node.name"


Summary:
Make observer fqn in prepare step independent of input_node/observed_node name.
This change names the observers as `{input/output}_activation_post_process_{idx}` where idx will be incremented for each new observer instance and is guaranteed to be unique.

Test Plan:
python test/test_quantization.py test_observer_fqn

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31752052](https://our.internmc.facebook.com/intern/diff/D31752052)

[ghstack-poisoned]
